### PR TITLE
[FLINK-16748][python][doc] Add Python UDTF doc

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request will add Python UDTF related doc*

## Brief change log

  - *Add doc description of Python UDTF in python_udfs.md and python_udfs.zh.md *
  - *Add some Python UDTF content in udfs.md and python_udfs.zh.md*
  - *Move the ussage of Java UDTF in Python from udfs.md to python_udfs.md*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: docs/dev/table/functions/udfs.md [hunk 1]
```diff
@@ -247,53 +253,51 @@ tableEnv.sqlQuery("SELECT a, word, length FROM MyTable LEFT JOIN LATERAL TABLE(s
 </div>
 
 <div data-lang="python" markdown="1">
-{% highlight python %}
-'''
-Java code:
+In order to define a Python table function, one can extend the base class TableFunction in pyflink.table.udtf and Implement an evaluation method. The behavior of a Python table function is determined by the evaluation method which is named eval.
```
Comment (hequn8128) [inline docs/dev/table/functions/udfs.md:256]: class \`TableFunction\` in \`pyflink.table.udtf\`

File: docs/dev/table/functions/udfs.md [hunk 2]
```diff
@@ -314,6 +318,12 @@ public class CustomTypeSplit extends TableFunction<Row> {
 </div>
 
 <div data-lang="scala" markdown="1">
+Please note that POJO types do not have a deterministic field order. Therefore, you cannot rename the fields of POJO returned by a table function using `AS`.
```
Comment (hequn8128) [inline docs/dev/table/functions/udfs.md:321]: Unnecessary changes.

File: docs/dev/table/functions/udfs.zh.md [hunk 3]
```diff
@@ -247,53 +255,51 @@ tableEnv.sqlQuery("SELECT a, word, length FROM MyTable LEFT JOIN LATERAL TABLE(s
...
+
+# use the Python Table function in SQL API
 table_env.sql_query("SELECT a, word, length FROM MyTable, LATERAL TABLE(split(a)) as T(word, length)")
-# LEFT JOIN a table function (equivalent to "left_outer_join" in Table API).
 table_env.sql_query("SELECT a, word, length FROM MyTable LEFT JOIN LATERAL TABLE(split(a)) as T(word, length) ON TRUE")
+
 {% endhighlight %}
+
+There are many ways to define a Python table function besides extending the base class `TableFunction`.
+Please refer to the [Python Table Function]({{ site.baseurl }}/dev/table/python/python_udfs.html#table-functions) documentation for more details.
```
Comment (hequn8128) [inline docs/dev/table/functions/udfs.zh.md:290]: Change the link to the Chinese doc link, i.e., `zh/dev/`

File: docs/dev/table/functions/udfs.zh.md [hunk 4]
```diff
@@ -314,6 +320,12 @@ public class CustomTypeSplit extends TableFunction<Row> {
 </div>
 
 <div data-lang="scala" markdown="1">
+Please note that POJO types do not have a deterministic field order. Therefore, you cannot rename the fields of POJO returned by a table function using `AS`.
```
Comment (hequn8128) [inline docs/dev/table/functions/udfs.zh.md:323]: Unnecessary change.

File: docs/dev/table/python/python_udfs.md [hunk 5]
```diff
@@ -125,3 +125,105 @@ table_env.register_function("add", add)
...
+      result = [1, 2, 3]
+      return result
+
+# option 2: return iterator
+@udtf(input_types=DataTypes.BIGINT(), result_types=DataTypes.BIGINT())
+def iterator_func(x):
+      return range(5)
+
+# option 3: generator function
+@udtf(input_types=DataTypes.BIGINT(), result_types=DataTypes.BIGINT())
```
Comment (hequn8128) [inline docs/dev/table/python/python_udfs.md:220]: Move this as option1, it can avoid creating temporary buffers, thus should be preferred. What do you think?

**Comment by HuangXingBo** [comment]
Thanks a lot for @hequn8128 review, I have addressed the comments at the latest commit.

**Comment by hequn8128** [comment]
@HuangXingBo Thanks a lot for the update. Merging...
